FirebaseInstallations/Source/Library/FIRInstallations.m

```diff
@@ -117,12 +119,40 @@ - (instancetype)initWithAppOptions:(FIROptions *)appOptions
   return self;
 }
 
++ (void)validateAppOptions:(FIROptions *)appOptions appName:(NSString *)appName {
+  NSMutableArray *missingFields = [NSMutableArray array];
+  if (appName.length < 1) {
+    [missingFields addObject:@"`FirebaseApp.name`"];
+  }
+  if (appOptions.APIKey.length < 1) {
+    [missingFields addObject:@"`FirebaseOptions.APIKey`"];
+  }
+  if (appOptions.googleAppID.length < 1) {
+    [missingFields addObject:@"`FirebaseOptions.googleAppID`"];
+  }
+  if (appOptions.GCMSenderID.length < 1) {
+    [missingFields addObject:@"`FirebaseOptions.GCMSenderID`"];
```

Comment:
Just to confirm - we need both the `googleAppID` *and* `gcmSenderID` right?

Reply:
As far as I know, both AppID and SenderID are required parameters to initialize FIROptions.
FIS, however, is interested in AppID, ProjectID, and ApiKey (and not necessarily in SenderID).

Reply:
@Maksym:
Why are you checking for Sender-ID?
We need either Project-ID (preferred) or Sender-ID, but we only need one of the two.

Reply:
We do NOT necessarily need GCMSenderID.
We only need GCMSenderID if projectID is not set.
Thus, I would recommend to change the check to look something like this:

if (appOptions.projectID.length < 1 && appOptions.GCMSenderID.length < 1) {
  [missingFields addObject:@"`FirebaseOptions.projectID`"];

Reply:
Updated and replied to the similar comment above.